lib/const.rb

```diff
@@ -115,8 +115,14 @@
   CREW_BUILD = 'x86_64-cros-linux-gnu'
 end
 
-CREW_COMMON_FLAGS = '-O2 -pipe -flto -ffat-lto-objects -fPIC -fuse-ld=gold'
-CREW_COMMON_FNO_LTO_FLAGS = '-O2 -pipe -fno-lto -fPIC -fuse-ld=gold'
+case ARCH
+when 'aarch64', 'armv7l'
+  CREW_LINKER = 'gold'
```

Comment:
Enable multithread support for `gold` by appending the following to `*FLAGS`? (theoretically would be slightly faster, but I don't see any difference in my system😅, but should be better than none)
```
-Wl,--threads -Wl,--thread-count,8
```

Reply:
Like this:
```ruby
unless ENV['CREW_LINKER'].to_s.empty?
  case ARCH
  when 'aarch64', 'armv7l'
    CREW_LINKER = 'gold'
    CREW_LINKER_FLAGS = "-Wl,--threads -Wl,--thread-count,#{CREW_NPROC}"
  when 'i686', 'x86_64'
    CREW_LINKER = 'mold'
    CREW_LINKER_FLAGS = ''
  end
else
  CREW_LINKER = ENV['CREW_LINKER']
end

CREW_COMMON_FLAGS = "-O2 -pipe -flto -ffat-lto-objects -fPIC -fuse-ld=#{CREW_LINKER} #{CREW_LINKER_FLAGS}"
CREW_COMMON_FNO_LTO_FLAGS = "-O2 -pipe -fno-lto -fPIC -fuse-ld=#{CREW_LINKER} #{CREW_LINKER_FLAGS}"
CREW_LDFLAGS = "-flto #{CREW_LINKER_FLAGS}"
CREW_FNO_LTO_LDFLAGS = '-fno-lto'
```

Reply:
Do those work with llvm?

If not, that could cause problems with packages which use clang.

Reply:
Yes, according to the reference of `clang`:
```
-Wl,<arg>,<arg2>...[¶](https://clang.llvm.org/docs/ClangCommandLineReference.html#cmdoption-clang-wl-arg-arg2)
Pass the comma separated arguments in <arg> to the linker
```

Reply:
I get this error on i686:
```
/usr/local/lib/crew/lib/const.rb:131:in `<top (required)>': uninitialized constant CREW_LINKER_FLAGS (NameError)                                                                CREW_COMMON_FLAGS = "-O2 -pipe -flto -ffat-lto-objects -fPIC -fuse-ld=#{CREW_LINKER} #{CREW_LINKER_FLAGS}"
                                                                                       ^^^^^^^^^^^^^^^^^
Did you mean?  CREW_LIB_PATH
        from /usr/local/bin/crew:12:in `require_relative'
        from /usr/local/bin/crew:12:in `<main>'
```
Why is setting `CREW_LINKER_FLAGS` to an empty string causing a problem?

Reply:
We should also set the flags under the `CREW_LINKER = ENV['CREW_LINKER']` part😅

Reply:
Actually, this was my fault. I had the logic backward!